### ISSUE TITLE
Conv with permutation on MPS will lead to negative MSE loss

### ISSUE THREAD

**Issue by Zhangyanbo** [pytorch/pytorch#124621]
### 🐛 Describe the bug

Repeat:

```python
import torch
import torch.nn as nn

lf = nn.MSELoss(reduction='none')
device = 'mps'

model = nn.Sequential(
    nn.Conv1d(3, 3, 1),
)

model = model.to(device)

x = torch.randn(128, 10, 3).to(device)
x = x.permute(0, 2, 1)
y = model(x)
y = y.permute(0, 2, 1)[:,:5,:]
y_hat = torch.randn(128, 5, 3).to(device)

loss = lf(y, y_hat)

if (loss < 0).any():
    print('negative')

# output: negative
```

> Note: you may need to run this code multiple times to see negative loss.

This works on CPU:

```python
import torch
import torch.nn as nn

lf = nn.MSELoss(reduction='none')
device = 'cpu'

model = nn.Sequential(
    nn.Conv1d(3, 3, 1),
)

model = model.to(device)

x = torch.randn(128, 10, 3).to(device)
x = x.permute(0, 2, 1)
y = model(x)
y = y.permute(0, 2, 1)[:,:5,:]
y_hat = torch.randn(128, 5, 3).to(device)

loss = lf(y, y_hat)

if (loss < 0).any():
    print('negative')

# no negative report
```

### Versions

PyTorch version: 2.2.2
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.2.1 (arm64)
GCC version: Could not collect
Clang version: 17.0.6
CMake version: version 3.26.2
Libc version: N/A

Python version: 3.8.19 | packaged by conda-forge | (default, Mar 20 2024, 12:49:57)  [Clang 16.0.6 ] (64-bit runtime)
Python platform: macOS-14.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1

Versions of relevant libraries:
[pip3] numpy==1.24.4
[pip3] torch==2.2.2
[pip3] torchaudio==2.2.2
[pip3] torchvision==0.17.2
[conda] numpy                     1.24.4           py38ha84db1f_0    conda-forge
[conda] pytorch                   2.2.2                   py3.8_0    pytorch
[conda] torchaudio                2.2.2                  py38_cpu    pytorch
[conda] torchvision               0.17.2                 py38_cpu    pytorch

cc @ezyang @gchanan @zou3519 @kadeng @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by Zhangyanbo** [comment]
This issue can be resolved by adding `.contiguous()` after permutation. But this is still an unexpected behavior.

**Comment by Zhangyanbo** [comment]
I feel it is more likely related to tensor shapes, because it can be fixed by adding `.contiguous()` after permutation. Another evidence is the `Conv1d`. The permutation only lead to this bug with convolution, but not itself alone.

**Comment by kulinseth** [comment]
@DenisVieriu97  has a fix for it which should address this issue. We are going through CI and should be able to merge it soon.
